bindings/rust/s2n-tls/src/callbacks.rs

```diff
@@ -17,220 +17,45 @@
 //! * "sync" callbacks return an immediate result and will block the task
 //!   performing the handshake until they return success or failure. See
 //!   [`VerifyHostNameCallback`] as an example.
-//! * "async" callbacks return a [`Poll`] and should not block the task performing the handshake.
-//!   They will be polled until they return [`Poll::Ready`].
+//! * "async" callbacks return a [Poll](`core::task::Poll`) and should not block the task performing the handshake.
+//!   They will be polled until they return [Poll::Ready](`core::task::Poll::Ready`).
 //!   [Connection::waker()](`crate::connection::Connection::waker()`)
 //!   can be used to register the task for wakeup. See [`ClientHelloCallback`] as an example.
 
-use crate::{
-    config::Config,
-    connection::{Connection, InternalConnectionFuture},
-    enums::CallbackResult,
-    error::Error,
-};
-use core::{mem::ManuallyDrop, ptr::NonNull, task::Poll, time::Duration};
-use pin_project_lite::pin_project;
+use crate::{config::Context, connection::Connection};
+use core::{mem::ManuallyDrop, ptr::NonNull, time::Duration};
 use s2n_tls_sys::s2n_connection;
-use std::{future::Future, pin::Pin};
+
+mod async_cb;
+pub use async_cb::*;
+
+mod client_hello;
+pub use client_hello::*;
 
 /// Convert the connection pointer provided to a callback into a Connection
-/// useable with the Rust bindings.
+/// and Context useable with the Rust bindings.
```

Comment:
Nit
```suggestion
/// and Context usable with the Rust bindings.
```

Reply:
Oh no, TIL "useable" is the British spelling. I've been spelling it "wrong" for a while D:

I can do a find/replace in the rest of the repo lol.